### ISSUE TITLE
just a question about frequency of  .DAILY

### ISSUE THREAD

**Issue by Photogad** [eggheadgames/Siren#47]
I just want to know exactly how often the .DAILY checks (or WHEN it checks).

My issue is the following:

I have the checker set to .IMMEDIATELY right now. I store the version.json file on my github in /docs folder and commit it to the repo when I'm ready to release the update. The problem with this is that it causes people in my app to get the update message before the update is approved by Google Play (takes  a few hours). 

SO I thought .DAILY would be better. But, when I had it set to .DAILY, not matter how often I opened the app, it never seemed to notify me that there was an update. (this is why I set it to .IMMEDIATELY).

So what exactly is the frequency that triggers the .DAILY setting to display the dialog to the users of the app?

**Comment by mikemee** [comment]
The "when to check" code is [here](https://github.com/eggheadgames/Siren/blob/91e99b5d3898e3b3cd4bd007ab39b53a3c558310/library/src/main/java/com/eggheadgames/siren/Siren.java#L98):

```java
        if (versionCheckType == SirenVersionCheckType.IMMEDIATELY) {
            performVersionCheck(appDescriptionUrl);
        } else if (versionCheckType.getValue() <= getSirenHelper().getDaysSinceLastCheck(mApplicationContext)
                ||getSirenHelper().getLastVerificationDate(mApplicationContext) == 0) {
            performVersionCheck(appDescriptionUrl);
    }
```
and `getDaysSinceLastCheck()` is [here](https://github.com/eggheadgames/Siren/blob/91e99b5d3898e3b3cd4bd007ab39b53a3c558310/library/src/main/java/com/eggheadgames/siren/SirenHelper.java#L42):

```java
    int getDaysSinceLastCheck(Context context) {
        long lastCheckTimestamp = getLastVerificationDate(context);

        if (lastCheckTimestamp > 0) {
            return (int) (TimeUnit.MILLISECONDS.toDays(Calendar.getInstance().getTimeInMillis()) - TimeUnit.MILLISECONDS.toDays(lastCheckTimestamp));
        } else {
            return 0;
        }
}
```
This uses a preference that resolves to the name `last_check_date`.

So, TL;DR, a setting of `DAILY` _should_ fire at the first Siren check that is 24 hours past the last time the app did a Siren check.  (Of course when that Siren check happens is up to your app).

Does this help explain what you're seeing?

**Comment by mikemee** [comment]
FWIW, I usually update the `json` file a couple of days after release, after I'm certain the new version isn't causing problems for the early/automatic updaters. So that might explain why I haven't had this particular experience, but it definitely makes sense.

**Comment by mikemee** [comment]
(Closed due to lack of response, and I think the explanation is complete...)
